Update locales-da-DK.xml

NEWille opened the pull request:
Corrected according to official Danish rules for writing names of months. Terms not translated from English corrected. Mistranslations corrected. Plural forms corrected.

rmzelle commented on locales-da-DK.xml:
```diff
@@ -54,18 +54,18 @@
-    <term name="open-inner-quote">‘</term>
-    <term name="close-inner-quote">’</term>
+    <term name="open-quote">"</term>
+    <term name="close-quote">»"</term>
```
Looks like you made a typo here: `<term name="close-quote">»"</term>`

NEWille replied:
Yes, the quotes should be of the same type: " ... " , but not of the type >>  ... <<. See also the next question.

rmzelle replied:
Straight quotes are exceedingly rare, though. Are you sure? E.g. https://en.wikipedia.org/wiki/Quotation_mark shows guillemets (https://en.wikipedia.org/wiki/Guillemet) or curly quotes for Danish.

(cc @adam3smith)

rmzelle commented on locales-da-DK.xml:
```diff
@@ -54,18 +54,18 @@
+    <term name="open-quote">"</term>
+    <term name="close-quote">»"</term>
+    <term name="open-inner-quote">'</term>
+    <term name="close-inner-quote">'</term>
```
Are you sure these should be straight quotes, and not curly quotes like those used in https://github.com/citation-style-language/locales/blob/master/locales-en-US.xml?

NEWille replied:
Curly quotes are all right, but not available at my keyboard. The ones I used will be OK for most bibliographies in manuscript.

adam3smith replied:
You can copy the curly ones from the English locale or from Wikipedia. But yes, we definitely want curly ones. Straight quotes are ugly.

rmzelle replied:
> Straight quotes are ugly.

And almost always, typographically incorrect.

NEWille replied:
Please substitute curly codes. It is not really an issue for me, just a practical "solution" to the problem of input.

rmzelle replied:
Could you try to do this yourself? We can't edit incoming pull requests.

Just visit https://github.com/NEWille/locales/blob/patch-1/locales-da-DK.xml, click on the pencil icon, and make the desired changes. You should be able to copy and paste the curly quotes from https://github.com/citation-style-language/locales/blob/master/locales-en-US.xml

rmzelle commented:
Hi @NEWille, there shouldn't be any reason for you to close and reopen your pull request. What are you trying to accomplish?

Your changes look good, other than the two questions I posted above. See our first reply for instructions on how to update this pull request.

adam3smith commented on locales-da-DK.xml:
```diff
@@ -199,8 +199,8 @@
     <term name="director">
-      <single>director</single>
-      <multiple>directors</multiple>
+      <single>instruktør</single>
```
"director" refers to film director in English. That's probably not the right translation, is it?

NEWille replied:
I think my answer got lost, so I repeat it:

Directors in film and theatre are "instruktører" in Danish. Directors of musical works are "dirigenter". The ambiguity of the English term creates problems for Danish, but this appears difficukt to resolve.

adam3smith replied:
OK, that's correct then, thanks. It just sounded very close to the English instructor and I wanted to make sure. (dirigenter would be conductor in English, so no ambiguity there.)

NEWille commented:
Hi @rmzelle I simply got confused about the proper way to finish the Pull Request. Sorry.

rmzelle commented:
This comment has instructions on how to edit the pull request: https://github.com/citation-style-language/locales/pull/134#issuecomment-193794503. Let us know if you have trouble figuring it out.

NEWille commented:
@rmzelle Sorry for the bother I seem to have created in my confusion.

I did not want to close this process, but did not understand the consequences. Actually I wanted the changes to result in a new version of the locales file for Danish. I submitted a full translation of this 4 years ago as a "gist", and its seems to have been forgotten completely. I do not want this revision have the same destiny.

I have now corrected to quotes to the preferred curly ones, and corrected the term sub verbo to the form used in Denmark, sub voce with the abreviated form s.v..

Hopefully I have now got the hang of the process.

NEWille

adam3smith commented:
Looks good to me now. Will merge tonight unless @rmzelle has remaining concerns

rmzelle commented:
@NEWille, thanks for contributing!

> I did not want to close this process, but did not understand the consequences. Actually I wanted the changes to result in a new version of the locales file for Danish. I submitted a full translation of this 4 years ago as a "gist", and its seems to have been forgotten completely. I do not want this revision have the same destiny.

We would only have seen your gist if you created a GitHub issue in the CSL locales repository (https://github.com/citation-style-language/locales/issues) at the same time with a link to the gist.

NEWille commented:
@rmzelle

Thank you for your patience and for explaining what went wrong 4 years
ago.

I will go on to localize some of the better known styles into Danish
(APA, MLA, Chicago (Author Date), ISO 690) to supplement
locales-da-DK.xml. And to create a specific style for the Danish language
Media anf Communications Encyclopedia that I am working for. All to be
used with Zotero in the Danish version.

Yours
Niels Erik Wille (@NEWille)

----- Original meddelelse -----

> Fra: Rintze M. Zelle <notifications@github.com>
> Til: citation-style-language/locales <locales@noreply.github.com>
> Cc: Niels Erik Wille <newinfo@tdcadsl.dk>
> Dato: Ons, 09. mar 2016 15:38
> Emne: Re: [locales] Update locales-da-DK.xml (#134)
> 
> @NEWille [https://github.com/NEWille], thanks for contributing!
> 
>   I did not want to close this process, but did not understand the
>   consequences. Actually I wanted the changes to result in a new
>   version of the locales file for Danish. I submitted a full
>   translation of this 4 years ago as a "gist", and its seems to
>   have been forgotten completely. I do not want this revision have
>   the same destiny.
> 
> We would only have seen your gist if you created a GitHub issue in
> the CSL locales repository (
> https://github.com/citation-style-language/locales/issues [https://github.com/citation-style-language/locales/issues])
> at the same time with a link to the gist.
> 
> —
> Reply to this email directly or view it on GitHub [https://github.com/citation-style-language/locales/pull/134#issuecomment-194322587].


NEW Information
v/ Niels Erik Wille
Berberisvej 7
2640 Hedehusene

E-post: newinfo@tdcadsl.dk
WWW: http://www.new-info.dk

rmzelle commented:
Found it! https://gist.github.com/NEWille/2196540 :)

But yeah, GitHub is a huge site, and CSL is just one project among millions. Unless you are active within the CSL repository, we don't get notified of your actions on GitHub.

> I will go on to localize some of the better known styles into Danish
(APA, MLA, Chicago (Author Date), ISO 690)

As you might already know, CSL styles without a "default-locale" attribute can be used in multiple languages in Zotero, in which case CSL will do its best to translate the style to the chosen language. Sometimes it's possible to fully support multiple languages with a single CSL style file, by relying on the CSL locale files and on in-style `<locale/>` elements (see http://docs.citationstyles.org/en/stable/specification.html#locale). This is the preferred option, but sometimes you will run into limitations in the extent to which CSL can localize styles, in which case you will have to create a separate copy of the style for Danish.

Does that make sense? So we prefer it if you first try to localize these styles to Danish by correcting terms and grammar settings with an in-style `<locale xml:lang="da-DK"/>` element. Only switch to submitting a separate style if you find that you have to change code within the `<citation/>`, `<bibliography/>`, and `<macro/>` elements.

NEWille commented:
Yes, I understand that the best way is to rely on the general locale file
for Danish ( locales-da-DK.xml (#134)) which I have just revised, and
then to enter further corrections relevant to the individual styles (if
they are needed at all) as in-style <locale/> elements.

Actually after further reflections the only style that I must do is the
one for the encyclopedia, and that might not even be submitted to the
shared repository, but only to the contributors to the encyclopedia and
the publishers.

I found the Github facilities for revising the locales-file very nice,
and would like use it for my work on the encyclopedia style since it is
going to be based on APA or ISO 690 with some important differences. Are
there other options?

If this is the wrong communication channel, please advise me.

Yours
Niels Erik Wille


----- Original meddelelse -----

> Fra: Rintze M. Zelle <notifications@github.com>
> Til: citation-style-language/locales <locales@noreply.github.com>
> Cc: Niels Erik Wille <newinfo@tdcadsl.dk>
> Dato: Ons, 09. mar 2016 19:52
> Emne: Re: [locales] Update locales-da-DK.xml (#134)
> 
> Found it! https://gist.github.com/NEWille/2196540 [https://gist.github.com/NEWille/2196540]
> :)
> 
> But yeah, GitHub is a huge site, and CSL is just one project among
> millions. Unless you are active within the CSL repository, we don't
> get notified of your actions on GitHub.
> 
>   I will go on to localize some of the better known styles into
>   Danish
>   (APA, MLA, Chicago (Author Date), ISO 690)
> 
> As you might already know, CSL styles without a "default-locale"
> attribute can be used in multiple languages in Zotero, in which case
> CSL will do its best to translate the style to the chosen language.
> Sometimes it's possible to fully support multiple languages with a
> single CSL style file, by relying on the CSL locale files and on
> in-style <locale/> elements (see
> http://docs.citationstyles.org/en/stable/specification.html#locale [http://docs.citationstyles.org/en/stable/specification.html#locale]).
> This is the preferred option, but sometimes you will run into
> limitations in the extent to which CSL can localize styles, in which
> case you will have to create a separate copy of the style for Danish.
> 
> Does that make sense? So we prefer it if you first try to localize
> these styles to Danish by correcting terms and grammar settings with
> an in-style <locale xml:lang="da-DK"/> element. Only switch to
> submitting a separate style if you find that you have to change code
> within the <citation/>, <bibliography/>, and <macro/> elements.
> 
> —
> Reply to this email directly or view it on GitHub [https://github.com/citation-style-language/locales/pull/134#issuecomment-194448688].


NEW Information
v/ Niels Erik Wille
Berberisvej 7
2640 Hedehusene

E-post: newinfo@tdcadsl.dk
WWW: http://www.new-info.dk

adam3smith commented:
sorry, I don't quite understand the question. Other options to do what precisely?

rmzelle commented:
> I found the Github facilities for revising the locales-file very nice,
and would like use it for my work on the encyclopedia style since it is
going to be based on APA or ISO 690 with some important differences. Are
there other options?

Yeah, it's not clear to me either what you're asking for. Are you looking for an editor, a way to keep track of changes, or something else?

NEWille commented:
I think what I am looking for is an editor with facilities for validating
my attempts at a new style - validitating as an CSL-file - but not
necssarily adding it to the styles repository.

Yours

Niels Erik Wille

----- Original meddelelse -----

> Fra: Rintze M. Zelle <notifications@github.com>
> Til: citation-style-language/locales <locales@noreply.github.com>
> Cc: Niels Erik Wille <newinfo@tdcadsl.dk>
> Dato: Tor, 10. mar 2016 19:05
> Emne: Re: [locales] Update locales-da-DK.xml (#134)
> 
>   I found the Github facilities for revising the locales-file very
>   nice,
>   and would like use it for my work on the encyclopedia style since
>   it is
>   going to be based on APA or ISO 690 with some important
>   differences. Are
>   there other options?
> 
> Yeah, it's not clear to me either what you're asking for. Are you
> looking for an editor, a way to keep track of changes, or something
> else?
> 
> —
> Reply to this email directly or view it on GitHub [https://github.com/citation-style-language/locales/pull/134#issuecomment-194980087].


Niels Erik Wille
Universitetslektor (emeritus), cand.mag.
Berberisvej 7
2640 Hedehusene
Tlf. 46 56 26 32
Mobiltlf. 40 16 96 85
E-post: n.e.wille@tdcadsl.dk

adam3smith commented:
There are only a couple of options I'm aware of for live validation. Those are 
- emacs with nxml mode (I use this. disadvantage: emacs takes a bit getting used to)
- jedit with plugin (never used this, found it hard to grasp)
- Oxygen (disadvantage: for pay, but I think most XML professionals use that)

The alternative is to use either the code editor in the visual editor:
http://editor.citationstyles.org/codeEditor/
or Zotero's built-in editor:
https://www.zotero.org/support/dev/citation_styles/reference_test_pane
both of which don't validate, but give you live preview of changes
(and then validate using http://validator.citationstyles.org/ )

rmzelle commented:
The easiest way to validate your styles (and locale files) is with http://formatter.citationstyles.org/. Personally I just edit styles in a plain text editor like Atom (https://atom.io/), Notepad++ (https://notepad-plus-plus.org/, Windows-only), or TextWrangler (http://www.barebones.com/products/textwrangler/, Mac-only, but some people use a dedicated XML editor (although some, like oXygen's XML Editor, are rather pricey [https://www.oxygenxml.com/]).

Or use the Visual CSL Editor, like @adam3smith just said.

(edit: beat by a minute by @adam3smith!)

NEWille commented:
@rmzelle & @adam3smith 

Thanks for the advice. I now see a way forward and will proceed in my own speed outside this forum. 

Looking forward to further cooperation if and when I need to do a pull request again. Perhaps further anmendments on the locales-da-DK? I won't need to do it for the work on the specialised style for the encyclopedia, I see that..

Yours Niels Erik Wille

rmzelle commented:
Good luck!
